src/plugins/intel_cpu/tests/functional/shared_tests_instances/snippets/mha.cpp

```diff
@@ -137,6 +137,32 @@ INSTANTIATE_TEST_SUITE_P(smoke_Snippets_MHAEnforceBF16,
                                             ::testing::Values(CPUTestUtils::cpu_bf16_plugin_config)),
                          MHA::getTestCaseName);
 
+INSTANTIATE_TEST_SUITE_P(smoke_Snippets_MHAFP16_4D,
+                         MHA,
+                         ::testing::Combine(::testing::ValuesIn(transposedShape_4D(false)),
+                                            ::testing::ValuesIn(precision_fp16_if_supported(4)),
+                                            ::testing::ValuesIn(mha_infer_precision_fp16_if_supported()),
+                                            ::testing::ValuesIn({false, true}),
+                                            ::testing::Values(MHA::default_thread_count),
+                                            ::testing::Values(2),
+                                            ::testing::Values(1),
+                                            ::testing::Values(ov::test::utils::DEVICE_CPU),
+                                            ::testing::Values(CPUTestUtils::empty_plugin_config)),
+                         MHA::getTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(smoke_Snippets_MHAEnforceFP16,
+                         MHA,
+                         ::testing::Combine(::testing::ValuesIn(transposedShape_4D(false)),
+                                            ::testing::ValuesIn(precision_f32(4)),
+                                            ::testing::ValuesIn(mha_infer_precision_fp16_if_supported()),
```

Comment:
```suggestion
                                            ::testing::ValuesIn(ov::element::f16),
```
And just skip test in `skip_test_config` if there is no ISA `amx_fp16` on the platform as it's implemented for bf16

Reply:
done, thanks Alexandra